packages/synapse-react-client/src/components/EntityTreeTable/hooks/useSorting.ts

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { SortingState } from '@tanstack/react-table'
+import { SortBy, Direction } from '@sage-bionetworks/synapse-types'
+
+export const useSorting = (
+  sorting: SortingState,
+  resetTreeData: () => void,
+) => {
+  // Track previous sorting to detect changes
+  const prevSortingRef = useRef<SortingState>([])
+  const isInitialMount = useRef(true)
+
+  // Convert TanStack Table sorting to API sorting parameters
+  const getSortingParams = useCallback(() => {
+    if (!sorting.length) return {}
+
+    const firstSort = sorting[0]
+    let sortBy: SortBy | undefined
+
+    switch (firstSort.id) {
+      case 'name':
+        sortBy = SortBy.NAME
+        break
+      case 'createdOn':
+        sortBy = SortBy.CREATED_ON
+        break
+      case 'modifiedOn':
+        sortBy = SortBy.MODIFIED_ON
+        break
+      default:
+        return {}
+    }
+
+    return {
+      sortBy,
+      sortDirection: firstSort.desc ? Direction.DESC : Direction.ASC,
+    }
+  }, [sorting])
+
+  // Reset tree when sorting changes (but not on initial mount)
+  useEffect(() => {
+    const sortingChanged =
+      JSON.stringify(prevSortingRef.current) !== JSON.stringify(sorting)
+
+    if (sortingChanged && !isInitialMount.current) {
+      // Reset tree data when sorting changes to force re-fetch with new sort order
+      // Don't reset sorting itself since that would clear the user's selection
+      resetTreeData()
+    }
+
+    prevSortingRef.current = sorting
+    isInitialMount.current = false
+  }, [sorting, resetTreeData])
```

Comment:
Basically, we need to ensure the tree state and sort state are synchronized. Remove this effect and put a more simple `useEffect` in `useTreeState` (captured in a comment on `useTreeState.ts`)